src/Exception/HttpExceptionTrait.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Superbrave\VerboseErrorHttpClient\Exception;
+
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Throwable;
+
+/**
+ * @author Niels Nijens <nn@superbrave.nl>
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+trait HttpExceptionTrait
+{
+    /**
+     * @var ResponseInterface
+     */
+    private $response;
+
+    /**
+     * @param ResponseInterface $response
+     * @param Throwable         $previousException
+     */
+    public function __construct(ResponseInterface $response, Throwable $previousException)
+    {
+        $this->response = $response;
+
+        $code = $response->getInfo('http_code');
+        $url = $response->getInfo('url');
+        $message = sprintf('HTTP %d returned for "%s".', $code, $url);
+
+        $httpCodeFound = false;
+        $isJson = false;
+        foreach (array_reverse($response->getInfo('response_headers')) as $h) {
```

Comment:
$h is not a very descriptive variable name..

Reply:
$responseHeader